radix-engine/src/blueprints/resource/auth_zone_composition.rs

```diff
@@ -211,22 +212,17 @@ fn max_ids_locked<Y: KernelSubstateApi + ClientApi<RuntimeError>>(
             )?;
             let proof: &NonFungibleProof = api.kernel_get_substate_ref(handle)?;
             for (container_id, locked_ids) in &proof.evidence {
-                let new_ids = locked_ids.clone();
-                if let Some(ids) = max.get_mut(container_id) {
-                    ids.extend(new_ids);
+                total.extend(locked_ids.clone());
+                if let Some(ids) = per_container.get_mut(container_id) {
+                    ids.extend(locked_ids.clone());
                 } else {
-                    max.insert(container_id.clone(), new_ids);
+                    per_container.insert(container_id.clone(), locked_ids.clone());
                 }
             }
         } else {
             api.sys_drop_lock(handle)?;
         }
     }
-    let mut total = BTreeSet::<NonFungibleLocalId>::new();
-    for value in max.values() {
```

Comment:
_Note from author:_

AFAIU, this post-iteration can be avoided (by collecting on the way), and I did that - please review carefully.